File: android/app/src/main/java/com/liive/ride/designsystem/LiiveDriverCard.kt
//  LiiveDriverCard.kt  ·  Liive Ride DS (Compose)  ·  mirrors components/ride/DriverCard
//  Composes LiiveAvatar + LiiveRatingStars.
package com.liive.ride.designsystem

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.painter.Painter
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.sp

@Composable
fun LiiveDriverCard(
    name: String,
    rating: Double? = null,
    vehicle: String? = null,
    plate: String? = null,
    eta: String? = null,
    speaking: Boolean = false,
    avatarPainter: Painter? = null,
    trailing: @Composable (() -> Unit)? = null,
) {
    val c = LiiveTheme.colors
    Row(
        Modifier
            .shadow(LiiveElevation.card, LiiveRadius.lg)
            .clip(LiiveRadius.lg)
            .background(c.surface)
            .padding(LiiveDriverCardLayout.CardPadding),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(LiiveDriverCardLayout.RowSpacing)
    ) {
        LiiveAvatar(name = name, size = LiiveDriverCardLayout.AvatarSize, ring = speaking, image = avatarPainter)
        Column(Modifier.weight(1f)) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(LiiveDriverCardLayout.TitleRatingSpacing)
            ) {
                Text(
                    name,
                    color = c.text,
                    style = MaterialTheme.typography.titleMedium,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                    modifier = Modifier.weight(1f, fill = false)
                )
                if (rating != null) LiiveRatingStars(value = rating, size = LiiveDriverCardLayout.RatingStarSize)
            }
            if (vehicle != null || plate != null) {
                Row(Modifier.fillMaxWidth().padding(top = LiiveDriverCardLayout.SecondaryLineTopPadding)) {
                    if (vehicle != null) {
                        Text(
                            vehicle,
                            color = c.textSecondary,
                            style = LiiveSheetMeta,
                            maxLines = 1,
                            overflow = TextOverflow.Ellipsis,
                            modifier = Modifier.weight(1f, fill = false)
                        )
                    }
                    if (vehicle != null && plate != null) {
                        Text(" · ", color = c.textSecondary, style = LiiveSheetMeta)
                    }
                    if (plate != null) {
                        Text(
                            plate,
                            color = c.text,
                            fontWeight = FontWeight.SemiBold,
                            style = LiiveSheetMeta.copy(letterSpacing = LiiveDriverCardLayout.PlateLetterSpacing),
                            maxLines = 1
                        )
                    }
                }
            }
        }
        if (eta != null) {
            Column(
                horizontalAlignment = Alignment.End,
                verticalArrangement = Arrangement.spacedBy(LiiveDriverCardLayout.TextSpacing)
            ) {
                Text(eta, color = c.accent, style = MaterialTheme.typography.headlineMedium.tabularNumbers())
                Text("away", color = c.textSecondary, style = MaterialTheme.typography.labelSmall)
            }
        }
        trailing?.invoke()
    }
}

private object LiiveDriverCardLayout {
    val RowSpacing = LiiveSpacing.s + LiiveSpacing.xs2
    val TitleRatingSpacing = LiiveSpacing.s
    val TextSpacing = LiiveSpacing.xs2
    val AvatarSize = LiiveControl.lg + LiiveSpacing.xs2
    val RatingStarSize = LiiveSpacing.m + LiiveSpacing.xs2 / 2
    val SecondaryLineTopPadding = LiiveSpacing.xs2
    val PlateLetterSpacing = (LiiveSpacing.xs2.value / 4).sp
    val CardPadding = RowSpacing
}

File: Packages/RideSharingFeature/Sources/RideSharingFeature/DesignSystem/LiiveDriverCard.swift
//  LiiveDriverCard.swift  ·  Liive Ride DS (SwiftUI)
//  Matched-driver summary: avatar, name, rating, vehicle + plate, ETA pill.
//  Composes LiiveAvatar + LiiveRatingStars.

import SwiftUI

public struct LiiveDriverCard<Trailing: View>: View {
    let name: String
    var rating: Double? = nil
    var vehicle: String? = nil
    var plate: String? = nil
    var eta: String? = nil
    var speaking: Bool = false
    var avatarImage: Image? = nil
    let trailing: Trailing

    public init(name: String, rating: Double? = nil, vehicle: String? = nil, plate: String? = nil,
                eta: String? = nil, speaking: Bool = false, avatarImage: Image? = nil,
                @ViewBuilder trailing: () -> Trailing = { EmptyView() }) {
        self.name = name; self.rating = rating; self.vehicle = vehicle; self.plate = plate
        self.eta = eta; self.speaking = speaking; self.avatarImage = avatarImage; self.trailing = trailing()
    }

    public var body: some View {
        HStack(spacing: LiiveDriverCardLayout.rowSpacing) {
            LiiveAvatar(name: name, image: avatarImage, size: LiiveDriverCardLayout.avatarSize, ring: speaking)
            VStack(alignment: .leading, spacing: LiiveDriverCardLayout.textSpacing) {
                HStack(spacing: LiiveDriverCardLayout.titleRatingSpacing) {
                    Text(name)
                        .font(LiiveFont.headline)
                        .foregroundColor(LiiveColor.text)
                        .lineLimit(1)
                        .truncationMode(.tail)
                        .layoutPriority(1)
                    if let rating {
                        LiiveRatingStars(value: rating, size: LiiveDriverCardLayout.ratingStarSize)
                            .fixedSize(horizontal: true, vertical: false)
                    }
                }
                if vehicle != nil || plate != nil {
                    HStack(spacing: 0) {
                        if let vehicle {
                            Text(vehicle)
                                .foregroundColor(LiiveColor.textSecondary)
                                .lineLimit(1)
                                .truncationMode(.tail)
                        }
                        if vehicle != nil && plate != nil { Text(" · ").foregroundColor(LiiveColor.textSecondary) }
                        if let plate {
                            Text(plate)
                                .fontWeight(.semibold)
                                .tracking(LiiveDriverCardLayout.plateTracking)
                                .foregroundColor(LiiveColor.text)
                                .lineLimit(1)
                                .fixedSize(horizontal: true, vertical: false)
                        }
                    }
                    .font(LiiveFont.sheetMeta)
                    .lineLimit(1)
                    .padding(.top, LiiveDriverCardLayout.secondaryLineTopPadding)
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            if let eta {
                VStack(alignment: .trailing, spacing: LiiveDriverCardLayout.textSpacing) {
                    Text(eta).font(LiiveFont.title2.monospacedDigit()).foregroundColor(LiiveColor.accent)
                    Text("away").font(LiiveFont.caption2).foregroundColor(LiiveColor.textSecondary)
                }
            }
            trailing
        }
        .padding(LiiveDriverCardLayout.cardPadding)
        .background(LiiveColor.surface)
        .clipShape(RoundedRectangle(cornerRadius: LiiveRadius.lg, style: .continuous))
        .liiveShadow(.card)
    }
}

private enum LiiveDriverCardLayout {
    static let rowSpacing = LiiveSpacing.m + LiiveSpacing.xs2
    static let titleRatingSpacing = LiiveSpacing.s
    static let textSpacing = LiiveSpacing.xs2
    static let avatarSize = LiiveControl.xl - LiiveSpacing.xs2
    static let ratingStarSize = LiiveSpacing.m + LiiveSpacing.xs2 / 2
    static let secondaryLineTopPadding = LiiveSpacing.xs2
    static let plateTracking = LiiveSpacing.xs2 / 4
    static let cardPadding = rowSpacing
}
